### PR TITLE
changes for DU940P-BU

### PR DESCRIPTION
**Pull request by mzieg**
Not yet ready for review.

### CODE REVIEW

**Comment by mzieg** [comment]
Ready for review.  I honestly don't know why the Newton camera now works, when it didn't before.  For the most part, all I did was add improved error reporting, so I could _see_ what the internal errors were...and now it's working.  Still, this is better I think.

File: wasatch/AndorDevice.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ class AndorDevice(InterfaceDevice):
     ############################################################################
     """
 
-    SUCCESS = 20002             #!< see page 330 of Andor SDK documentation
-    SHUTTER_SPEED_MS = 35       #!< not sure where this comes from...ask Caleb - TS
+    SUCCESS = 20002             #!< see load_error_codes()
+    SHUTTER_SPEED_MS = 50       #!< empirically determined
```
Comment (mzieg) [inline wasatch/AndorDevice.py:51]: This may? be what got the Newton working

File: wasatch/AndorDevice.py [hunk 2]
```diff
@@ -293,31 +300,41 @@ def _take_one_averaged_reading(self) -> SpectrometerResponse:
...
-        log.info("initializing camera...")
+        self.check_result(self.driver.GetCameraHandle(self.spec_index, byref(cameraHandle)), "GetCameraHandle")
+        self.check_result(self.driver.SetCurrentCamera(cameraHandle.value), "SetCurrentCamera")
 
-        # not sure init_str is actually required
-        init_str = create_string_buffer(b'\000' * 16)
-        assert(self.SUCCESS == self.driver.Initialize(init_str)), "unable to initialize camera"
-        log.info("success")
+        try:
+            path_to_ini = create_string_buffer(b'\000' * 256)
```
Comment (mzieg) [inline wasatch/AndorDevice.py:333]: Increased this to 256 bytes because it's a Windows directory path

File: wasatch/AndorDevice.py [hunk 3]
```diff
@@ -372,6 +385,7 @@ def _load_config_values(self):
         # same spelling
         for k in [ 'detector', 
                    'model', 
+                   'detector',
```
Comment (mzieg) [inline wasatch/AndorDevice.py:388]: since not all Andor cameras are iDus, as we now know...

File: wasatch/WrapperWorker.py [hunk 4]
```diff
@@ -73,7 +73,7 @@ def run(self) -> None:
                     self.connected_device = self.device_id.device_type
                     self.connected_device.disconnect = False
                 else:
-                    self.connected_device = device_classes[3](
+                    self.connected_device = device_classes[type_connection](
```
Comment (mzieg) [inline wasatch/WrapperWorker.py:76]: Assuming that was the intent :-)

**Review by rc14193 — APPROVED** [review]

**Pull request merged**